qt/python/mantidqt/widgets/colorbar/colorbar.py

```diff
@@ -161,15 +167,20 @@ def norm_changed(self):
         Called when a different normalization is selected
         """
         idx = self.norm.currentIndex()
+        scale = NORM_OPTS[idx]
         if NORM_OPTS[idx] == 'Power':
             self.powerscale.show()
             self.powerscale_label.show()
+            if self.powerscale.hasAcceptableInput():
+                scale += "," + self.powerscale.text()
         else:
             self.powerscale.hide()
             self.powerscale_label.hide()
         self.colorbar.mappable.set_norm(self.get_norm())
         self.set_mappable(self.colorbar.mappable)
 
+        ConfigService.setString("sliceviewer.colorbar.scale", scale)
```

Comment:
I'm afraid this isn't quite the right place to do this. The `colorbar` is a reusable widget that is used in other places and thus can't assume what its host is. The sliceviewer is the user/client of this code so needs to be the one to decide what to save.

Is this something we want to be controlled from the properties file and preset or do we just want to remember what it did last time? I think the latter applies more in this situation at the moment when we just want persistence of the last value. There is a model of how this is done with the project saving that accepts an object to access the session configuration. I think the SliceViewer presenter should be able to be updated in a similar style.

Reply:
Following on from the discussion in the other comments. Can we update this to do something similar to the [saveprojectdialog](https://github.com/mantidproject/mantid/blob/master/qt/python/mantidqt/widgets/saveprojectdialog/presenter.py#L19). The presenter accepts a `conf` instance that allows it to save something in the conf object of the client, e.g workbench. I think we can have the SliceViewer presenter do something similar and save to the conf object. As Andrei says maybe a key like `sliceviewer/scale_type` so it's clear what it's referring to.

Let me know if you want to chat over anything about this.